opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -128,170 +128,183 @@ class SimulatorFullyImplicitBlackoilEbos
     /// \param[in,out] state       state of reservoir: pressure, fluxes
     /// \return                    simulation report, with timing data
     SimulatorReport run(SimulatorTimer& timer)
+    {
+        runInit(timer);
+        // Main simulation loop.
+        while (!timer.done()) {
+            runStep(timer);
```

Comment:
I think you should check the return value from `runStep()` and return if necessary.